### PR TITLE
fix: Correct failing municipality integration test

### LINKED ISSUE #319: Integration test failing on feeds endpoint filtering by municipality
### Describe the bug

Integration tests are failing in QA and PROD while running `GTFSFeedsEndpointTests.test_filter_by_municipality_gtfs` with the following error:
```
AssertionError: Expected all feeds to have municipality 'Napoli', but found 
municipalities '['Indianapolis']'
```
The artifact of the integration test results has not been generated, making it hard to get the full report locally for further analysis.

Example failing build, [link](https://github.com/MobilityData/mobility-feed-api/actions/runs/8074913508/job/22078412827)
Current API version:
```
{
    "version": "v0.0.0",
    "commit_hash": "2c9d680b6547a180b1d3d4e9f2ccd88872d920c7"
}
```

### Steps/Code to Reproduce

- Execute integration test on the QA environment
- Verify that it is failing
- Verify that the integration test results artifact is not uploaded 

### Expected Results

- Integration test passed
- The integration test results artifact is generated even if the job fails

### Actual Results

- Integration test passed
- The integration test results artifact is not generated when the job fails

### Screenshots

_No response_

### Files used

_No response_

### Additional notes

_No response_

### PR DESCRIPTION
**Pull request by jcpitre**
**Summary:**
Closes #319 

Integration test for municipalities was failing.

**Expected behavior:** 

They should not fail.

**Testing tips:**

Run the integration tests

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by jcpitre** [comment]
Question: I don't understand why tihs error just shows up now if we ran the tests countless times before.

**Comment by davidgamez** [comment]
@jcpitre I'm ready to approve this PR if the requirement _"The integration test results artifact is generated even if the job fails"_; will be addressed in a follow-up PR

File: integration-tests/src/endpoints/integration_tests.py [hunk 1]
```diff
@@ -108,12 +108,16 @@ def _test_filter_by_municipality(
...
+        lowercase_municipality = municipality.lower()
         if validate_location:
             for feed in feeds:
                 municipalities = [
                     location["municipality"] for location in feed["locations"]
                 ]
-                is_municipality_valid = [municipality in m for m in municipalities]
+                is_municipality_valid = [
+                    lowercase_municipality in m.lower() for m in municipalities
+                ]
```
Comment (davidgamez) [inline integration-tests/src/endpoints/integration_tests.py:120]: 👍

**Comment by davidgamez** [comment]
> Question: I don't understand why tihs error just shows up now if we ran the tests countless times before.

We had been actively adding new feeds, it might be related to new feeds containing municipalities with lower/upper-case differences.

**Review by davidgamez — APPROVED** [review]

**Pull request merged**